edit_questionpy_form.php

```diff
@@ -70,6 +71,81 @@ public function __construct(string $submiturl, object $question, object $categor
         parent::__construct($submiturl, $question, $category, $contexts, $formeditable);
     }
 
+
+    /**
+     * Returns the question state of the stored question if possible.
+     *
+     * If the question is being edited and the package (version) was changed, no question state is returned.
+     *
+     * @param string $packagehash
+     * @return string|null
+     */
+    private function get_question_state(string $packagehash): ?string {
+        if (isset($this->question->qpy_id) && $this->question->qpy_package_hash === $packagehash) {
+            return $this->question->qpy_state;
+        }
+        return null;
+    }
+
+    /**
+     * Adds alert to the form to warn that the selected package is not usable even though it is available through the package
+     * search.
+     *
+     * @param MoodleQuickForm $mform
+     * @param string $hash
+     * @return void
+     * @throws moodle_exception
+     */
+    private function definition_package_not_usable(MoodleQuickForm $mform, string $hash): void {
+        global $OUTPUT;
+
+        // The current package is in use if the current question is saved and its package hash matches the hash of the selected
+        // package.
+        $packageinuse = isset($this->question->qpy_id) && $this->question->qpy_package_hash == $hash;
```

Comment:
Ich würde der Einfachheit halber einfach immer "[Existing ]questions using this package will not work." anzeigen, auch wenn _diese_ Frage noch nicht mit dem kaputten Paket gespeichert wurde, ist das ja schon recht wichtig zu wissen.